tests/plotting/test_plotting.py

```diff
@@ -129,6 +144,10 @@ def verify_cache_image(plotter):
         allowed_error = IMAGE_REGRESSION_ERROR
         allowed_warning = IMAGE_REGRESSION_WARNING
 
+    # some tests fail when on Windows with OSMesa
+    if os.name == 'nt' and item.function in WINDOWS_SKIP_IMAGE_CACHE:
```

Comment:
We could make this a bit prettier by putting this check (and the variance/regression if block right before) after the next block that raises on `test_name is None`, because then we could use `test_name` instead of `item.function` in several places.

(And probably a bit obscure but we could remove the `test_name = None` sentinel value and instead use a `for-else` construct in the previous loop to raise when `break` wasn't hit.)

Reply:
Both good points. Resolved in b0a6a86ab96e67c9deb5b4e610b605abc6ea832f.